compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -248,6 +249,18 @@ generatePackageRule =
         dalf <- use_ GenerateDalf file
         return ([], Just $ WhnfPackage $ deps{LF.packageModules = NM.insert dalf (LF.packageModules deps)})
 
+generateVisualizationRule :: Rules ()
```

Comment:
I don’t think adding a rule for visualization is a good idea:

Adding more rules is the usual memory vs time tradeoff. If you put it in a rule, you get caching so you don’t have to recompute it if nothing has changed. On the other hand you have to keep it in memory which increases memory usage.

In this case caching is quite poor: As soon as any file in the dependency tree of `file` has changed you will rerun the whole visualization. That means that the only case where we benefit from caching is if the user reruns the visualization without making any changes. That does not seem like a very common case that we should be optimizing for. Compare that to a rule like typechecking that reuses the typechecked modules so you gain a lot more from caching since while the user might have modified the current file at least some of the dependencies will not have changed so caching is crucial to give fast feedback on incremental changes.

Reply:
Ah ok that makes sense, I did not think it about memory consumption.